src/api.py

```diff
@@ -113,24 +133,35 @@ class GPT4PluginConfig(Config):
 
     @classmethod
     def config_cls(cls) -> Type[Config]:
-        return cls.GPT4PluginConfig
+        return cls.LiteLLMPluginConfig
 
-    config: GPT4PluginConfig
+    config: LiteLLMPluginConfig
 
     def __init__(
         self,
         client: Steamship = None,
         config: Dict[str, Any] = None,
         context: InvocationContext = None,
     ):
-        # Load original api key before it is read from TOML, so we know to restrict models for billing
-        original_api_key = config.get("openai_api_key", "")
+        # Load original env before it is read from TOML, so we know whether to bill usage or not.
+        original_env = config.get("litellm_env", "")
         super().__init__(client, config, context)
-        if original_api_key == "" and self.config.model not in VALID_MODELS_FOR_BILLING:
-            raise SteamshipError(
-                f"This plugin cannot be used with model {self.config.model} while using Steamship's API key. Valid models are {VALID_MODELS_FOR_BILLING}"
-            )
-        openai.api_key = self.config.openai_api_key
+        if config.get("n", 1) != 1:
+            raise SteamshipError("There is currently a known bug with this plugin and config value n != 1.")
+
+        self.apply_env(self.config.litellm_env)
+        self.steamship_billing = (original_env == "")
```

Comment:
If users provide their own litellm environment, we don't bill for them; they're providing their own keys.